src/strategies/soundxyz-metadata/extractor.mjs

```diff
@@ -0,0 +1,113 @@
+// @format
+import { env } from "process";
+import { encodeCallSignature, decodeCallOutput } from "eth-fun";
+
+export const name = "soundxyz-metadata";
+export const props = {
+  signatures: {
+    tokenToEdition: "tokenToEdition(uint256)",
+    editions: "editions(uint256)",
+  },
+  contract: {
+    address: "0x01ab7d30525e4f3010af27a003180463a6c811a6",
+  },
+  options: {
+    url: env.RPC_HTTP_HOST,
+    headers: {
+      Authorization: `Bearer ${env.RPC_API_KEY}`,
+    },
+  },
+};
+
+export function init(tokenId) {
+  const data = encodeCallSignature(
+    props.signatures.tokenToEdition,
+    ["uint256"],
+    [tokenId]
+  );
+
+  const from = null;
+  return {
+    messages: [
+      {
+        type: "json-rpc",
+        commissioner: name,
+        options: props.options,
+        version: "0.0.1",
+        method: "eth_call",
+        params: [
+          {
+            from,
+            to: props.contract.address,
+            data,
+          },
+          "latest",
```

Comment:
Is this approach used in PR #53 appropriate for here too? That is, along with tokenId we should pass block number. Since, that PR has not been approved I was reluctant to use the same here.

https://github.com/neume-network/strategies/blob/00ca252faa2b2ae0d79d9dcd2f55a5ac994fffd6/src/strategies/soundxyz/extractor.mjs#L23

Reply:
> Since, that PR has not been approved I was reluctant to use the same here.

Good call, I ended up requesting changes in #53. Please consider what was said in the requested changes when making the block number change here: https://github.com/neume-network/strategies/pull/53#pullrequestreview-989948228

Reply:
The changes requested in https://github.com/neume-network/strategies/pull/53#pullrequestreview-989948228 were not valid here. They are to be done from where this strategy will be called. Currently, it is not being called from anywhere. I am not sure about where it is to be called. Maybe from web3subgraph in parallel to soundxyz?

But I have modified the strategy to accept a block number and use it instead of latest.

Reply:
ok I'm fine with merging "invalid" code and e.g. creating respective issues to change it later.